File: examples/example_polygonal_smooth_certificate.ts
/**
 * polygonal_smooth_certificate_node.cpp — see examples/example_polygonal_smooth_certificate.py
 */
const sst = require('../index.js');

function circle(n, r = 1.0) {
    const pts = [];
    for (let i = 0; i < n; i++) {
        const t = (i / n) * 2 * Math.PI;
        pts.push([r * Math.cos(t), r * Math.sin(t), 0]);
    }
    return pts;
}

function main() {
    if (!sst.isAvailable || typeof sst.evaluatePolygonalSmoothCertificate !== 'function') {
        console.log('[SKIP] polygonal_smooth_certificate: addon not available');
        return;
    }
    const ring = circle(48);
    const cert = sst.evaluatePolygonalSmoothCertificate(ring, ring, 0.05, 1e-9, 1e-9, 1e-6);
    console.log('evaluatePolygonalSmoothCertificate:', cert.status, 'hausdorff=', cert.hausdorffBound);

    const aK = 1 / (4 * Math.PI);
    const biot = sst.evaluateBiotSavartGate(aK, 0, 1.0, 0.1, 128, 'desing_core_v1', 1e-12, true);
    console.log('evaluateBiotSavartGate:', biot.status, 'residual=', biot.relativeResidual);
}

main();

File: examples/example_polygonal_smooth_certificate.py
"""Smoke example for polygonal_smooth_certificate + Biot–Savart gate (Canon v0.8.21+)."""
from __future__ import annotations

import math
import os
import sys

sys.path.insert(0, os.path.abspath("."))
sys.path.insert(0, os.path.join(os.path.abspath("."), "src"))

try:
    from SSTcore import PolygonalSmoothCertificateAPI, BiotSavartGateAPI
except ImportError:
    from sstcore import PolygonalSmoothCertificateAPI, BiotSavartGateAPI


def circle(n: int = 48, radius: float = 1.0):
    return [
        [radius * math.cos(2.0 * math.pi * i / n), radius * math.sin(2.0 * math.pi * i / n), 0.0]
        for i in range(n)
    ]


pts = circle(48)
cert = PolygonalSmoothCertificateAPI.evaluate(
    pts, pts, tube_radius=0.05, hausdorff_tol=1e-9, tangent_tol=1e-9, curvature_tol=1e-6
)
print("polygonal_smooth evaluate:", cert.status, "hausdorff=", cert.hausdorff_bound)
print("fingerprint:", PolygonalSmoothCertificateAPI.fingerprint_points(pts))

a_k = 1.0 / (4.0 * math.pi)
biot = BiotSavartGateAPI.evaluate(
    observable=a_k,
    estimated_limit=0.0,
    boundary_margin=1.0,
    min_boundary_margin=0.1,
    sample_count=128,
    regularization_id="desing_core_v1",
    residual_tol=1e-12,
    use_four_pi_identity=True,
)
print("biot_savart_gate:", biot.status, "residual=", biot.relative_residual)
